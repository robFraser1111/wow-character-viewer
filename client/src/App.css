body {
    background: #15171e;
}

* {
    font-family: "Open Sans", Arial, Helvetica, sans-serif;
    margin: 0;
    padding: 0;
}

h1 {
    font-size: 28px;
}

h2 {
    font-size: 26px;
}

h3 {
    font-size: 24px;
}

h4 {
    font-size: 22px;
}

h5 {
    font-size: 20px;
}

h6 {
    font-size: 18px;
}

p {
    font-size: 16px;
}

a {
    text-decoration: none;
}

select,
option {
    cursor: pointer;
}

a,
button {
    transition: 0.2s;
    cursor: pointer;
}
