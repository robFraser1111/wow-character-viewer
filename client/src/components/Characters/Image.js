import React, { useContext } from "react";
import useFetch from "../../hooks/useFetch";
import UserContext from "../../store/context";
import Placeholder from "../../images/wow-placeholder-300-300-01.jpg";

const Image = (props) => {
    const user = useContext(UserContext);

    const currentImage = useFetch(props.characterImage, "GET", user.token);

    return (
        <>
            {currentImage.hasOwnProperty("assets") ? (
                <img src={currentImage.assets[2].value} />
            ) : (
                <img src={Placeholder} />
            )}
        </>
    );
};

export default Image;
