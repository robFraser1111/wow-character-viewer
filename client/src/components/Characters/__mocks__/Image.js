import React from "react";
import Placeholder from "../../../images/wow-placeholder-300-300-01.jpg";

const Image = () => {
    const currentImage = {
        _links: {
            self: {
                href: "https://us.api.blizzard.com/profile/wow/character/bathtub/bigdawg/character-media?namespace=profile-us",
            },
        },
        character: {
            key: {
                href: "https://us.api.blizzard.com/profile/wow/character/bathtub/bigdawg?namespace=profile-us",
            },
            name: "Bigdawg",
            id: 1,
            realm: {
                key: {
                    href: "https://us.api.blizzard.com/data/wow/realm/1?namespace=dynamic-us",
                },
                name: {
                    en_US: "Barthilas",
                    es_MX: "Barthilas",
                    pt_BR: "Barthilas",
                    de_DE: "Barthilas",
                    en_GB: "Barthilas",
                    es_ES: "Barthilas",
                    fr_FR: "Barthilas",
                    it_IT: "Barthilas",
                    ru_RU: "Barthilas US9",
                    ko_KR: "Barthilas",
                    zh_TW: "巴瑟拉斯",
                    zh_CN: "巴瑟拉斯",
                },
                id: 1337,
                slug: "barthilas",
            },
        },
        assets: [
            {
                key: "avatar",
                value: "https://render.worldofwarcraft.com/us/character/barthilas/69/1-avatar.jpg",
            },
            {
                key: "inset",
                value: "https://render.worldofwarcraft.com/us/character/barthilas/69/1-inset.jpg",
            },
            {
                key: "main",
                value: "https://render.worldofwarcraft.com/us/character/barthilas/69/1-main.jpg",
            },
            {
                key: "main-raw",
                value: "https://render.worldofwarcraft.com/us/character/barthilas/69/1-main-raw.png",
            },
        ],
    };

    return (
        <>
            {currentImage.hasOwnProperty("assets") ? (
                <img src={currentImage.assets[2].value} />
            ) : (
                <img src={Placeholder} />
            )}
        </>
    );
};

export default Image;
