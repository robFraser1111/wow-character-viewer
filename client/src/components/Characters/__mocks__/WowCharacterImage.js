import React from "react";
import Image from "./Image";
import Placeholder from "../../../images/wow-placeholder-300-300-01.jpg";
import styled from "styled-components";

const CharacterImage = styled.div`
    max-width: 300px;
    max-height: 300px;

    img {
        width: 300px;
        height: 300px;
    }
`;

const WowCharacterImage = () => {
    const currentCharacter = {
        _links: {
            self: {
                href: "https://us.api.blizzard.com/profile/wow/character/bathtub/bigdawg?namespace=profile-us",
            },
        },
        id: 1,
        name: "Bigdawg",
        gender: {
            type: "MALE",
            name: {
                en_US: "Male",
                es_MX: "Masculino",
                pt_BR: "Masculino",
                de_DE: "Männlich",
                en_GB: "Male",
                es_ES: "Masculino",
                fr_FR: "Homme",
                it_IT: "Maschio",
                ru_RU: "Мужской",
                ko_KR: "남성",
                zh_TW: "男性",
                zh_CN: "男性",
            },
        },
        faction: {
            type: "ALLIANCE",
            name: {
                en_US: "Alliance",
                es_MX: "Alianza",
                pt_BR: "Aliança",
                de_DE: "Allianz",
                en_GB: "Alliance",
                es_ES: "Alianza",
                fr_FR: "Alliance",
                it_IT: "Alleanza",
                ru_RU: "Альянс",
                ko_KR: "얼라이언스",
                zh_TW: "聯盟",
                zh_CN: "联盟",
            },
        },
        race: {
            key: {
                href: "https://us.api.blizzard.com/data/wow/playable-race/1?namespace=static-1-us",
            },
            name: {
                en_US: "Dwarf",
                es_MX: "Enano",
                pt_BR: "Anão",
                de_DE: "Zwerg",
                en_GB: "Dwarf",
                es_ES: "Enano",
                fr_FR: "Nain",
                it_IT: "Nano",
                ru_RU: "Дворф",
                ko_KR: "드워프",
                zh_TW: "矮人",
                zh_CN: "矮人",
            },
            id: 3,
        },
        character_class: {
            key: {
                href: "https://us.api.blizzard.com/data/wow/playable-class/1?namespace=static-1-us",
            },
            name: {
                en_US: "Paladin",
                es_MX: "Paladín",
                pt_BR: "Paladino",
                de_DE: "Paladin",
                en_GB: "Paladin",
                es_ES: "Paladín",
                fr_FR: "Paladin",
                it_IT: "Paladino",
                ru_RU: "Паладин",
                ko_KR: "성기사",
                zh_TW: "聖騎士",
                zh_CN: "圣骑士",
            },
            id: 2,
        },
        active_spec: {
            key: {
                href: "https://us.api.blizzard.com/data/wow/playable-specialization/1?namespace=static-1-us",
            },
            name: {
                en_US: "Protection",
                es_MX: "Protección",
                pt_BR: "Proteção",
                de_DE: "Schutz",
                en_GB: "Protection",
                es_ES: "Protección",
                fr_FR: "Protection",
                it_IT: "Protezione",
                ru_RU: "Защита",
                ko_KR: "보호",
                zh_TW: "防護",
                zh_CN: "防护",
            },
            id: 69,
        },
        realm: {
            key: {
                href: "https://us.api.blizzard.com/data/wow/realm/69?namespace=dynamic-us",
            },
            name: {
                en_US: "Barthilas",
                es_MX: "Barthilas",
                pt_BR: "Barthilas",
                de_DE: "Barthilas",
                en_GB: "Barthilas",
                es_ES: "Barthilas",
                fr_FR: "Barthilas",
                it_IT: "Barthilas",
                ru_RU: "Barthilas US9",
                ko_KR: "Barthilas",
                zh_TW: "巴瑟拉斯",
                zh_CN: "巴瑟拉斯",
            },
            id: 69,
            slug: "bathtub",
        },
        level: 69,
        experience: 69,
        achievement_points: 69,
        achievements: {
            href: "https://us.api.blizzard.com/profile/wow/character/bathtub/bigdawg/achievements?namespace=profile-us",
        },
        titles: {
            href: "https://us.api.blizzard.com/profile/wow/character/bathtub/bigdawg/titles?namespace=profile-us",
        },
        pvp_summary: {
            href: "https://us.api.blizzard.com/profile/wow/character/bathtub/bigdawg/pvp-summary?namespace=profile-us",
        },
        encounters: {
            href: "https://us.api.blizzard.com/profile/wow/character/bathtub/bigdawg/encounters?namespace=profile-us",
        },
        media: {
            href: "https://us.api.blizzard.com/profile/wow/character/bathtub/bigdawg/character-media?namespace=profile-us",
        },
        last_login_timestamp: 69,
        average_item_level: 69,
        equipped_item_level: 69,
        specializations: {
            href: "https://us.api.blizzard.com/profile/wow/character/bathtub/bigdawg/specializations?namespace=profile-us",
        },
        statistics: {
            href: "https://us.api.blizzard.com/profile/wow/character/bathtub/bigdawg/statistics?namespace=profile-us",
        },
        mythic_keystone_profile: {
            href: "https://us.api.blizzard.com/profile/wow/character/bathtub/bigdawg/mythic-keystone-profile?namespace=profile-us",
        },
        equipment: {
            href: "https://us.api.blizzard.com/profile/wow/character/bathtub/bigdawg/equipment?namespace=profile-us",
        },
        appearance: {
            href: "https://us.api.blizzard.com/profile/wow/character/bathtub/bigdawg/appearance?namespace=profile-us",
        },
        collections: {
            href: "https://us.api.blizzard.com/profile/wow/character/bathtub/bigdawg/collections?namespace=profile-us",
        },
        reputations: {
            href: "https://us.api.blizzard.com/profile/wow/character/bathtub/bigdawg/reputations?namespace=profile-us",
        },
        quests: {
            href: "https://us.api.blizzard.com/profile/wow/character/bathtub/bigdawg/quests?namespace=profile-us",
        },
        achievements_statistics: {
            href: "https://us.api.blizzard.com/profile/wow/character/bathtub/bigdawg/achievements/statistics?namespace=profile-us",
        },
        professions: {
            href: "https://us.api.blizzard.com/profile/wow/character/bathtub/bigdawg/professions?namespace=profile-us",
        },
    };

    return (
        <CharacterImage>
            {currentCharacter.hasOwnProperty("media") ? (
                <Image characterImage={currentCharacter.media.href} />
            ) : (
                <img src={Placeholder} />
            )}
        </CharacterImage>
    );
};

export default WowCharacterImage;
