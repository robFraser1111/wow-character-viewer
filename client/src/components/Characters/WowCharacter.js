import React, { useContext } from "react";
import useFetch from "../../hooks/useFetch";
import UserContext from "../../store/context";
import WowCharacterImage from "./WowCharacterImage";
import styled from "styled-components";

const Character = styled.div`
    display: flex;
    flex-direction: row;
    margin: 10px;
    width: 680px;
    background-color: ${(props) => props.theme.brown};
    border: 1px solid ${(props) => props.theme.yellowDark};

    @media (max-width: ${(props) => props.theme.tablet}) {
        width: 300px;
        flex-direction: column;
    }
`;

const Text = styled.div`
    padding: 20px;

    h3 {
        color: ${(props) => props.theme.yellow};
    }

    h4 {
        color: ${(props) => props.theme.yellowDark};
    }

    p {
        margin: 8px 0;
        color: ${(props) => props.theme.yellowLight};
    }
`;

const WowCharacter = (props) => {
    const user = useContext(UserContext);

    const currentCharacter = useFetch(
        `https://${user.region}.api.blizzard.com/profile/user/wow/protected-character/${props.realmId}-${props.characterId}?namespace=profile-${user.region}&locale=${user.locale}`,
        "GET",
        user.token
    );

    return (
        <Character>
            <WowCharacterImage
                character={
                    currentCharacter.hasOwnProperty("character") &&
                    currentCharacter.character.key.href
                }
            />
            <Text>
                <h3>{props.name}</h3>
                <h4>Realm: {props.name}</h4>

                <p>Level: {props.level}</p>

                <p>Class: {props.playableClass}</p>

                <p>Race: {props.playableRace}</p>

                <p>Gender: {props.gender}</p>

                <p>Faction: {props.faction}</p>
            </Text>
        </Character>
    );
};

export default WowCharacter;
