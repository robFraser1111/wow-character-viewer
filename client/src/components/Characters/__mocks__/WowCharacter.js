import React from "react";
import WowCharacterImage from "./WowCharacterImage";
import styled from "styled-components";

const Character = styled.div`
    display: flex;
    flex-direction: row;
    margin: 10px;
    width: 680px;
    background-color: ${(props) => props.theme.brown};
    border: 1px solid ${(props) => props.theme.yellowDark};

    @media (max-width: ${(props) => props.theme.tablet}) {
        width: 300px;
        flex-direction: column;
    }
`;

const Text = styled.div`
    padding: 20px;

    h3 {
        color: ${(props) => props.theme.yellow};
    }

    h4 {
        color: ${(props) => props.theme.yellowDark};
    }

    p {
        margin: 8px 0;
        color: ${(props) => props.theme.yellowLight};
    }
`;

const WowCharacter = (props) => {
    const currentCharacter = {
        _links: {
            self: {
                href: "https://us.api.blizzard.com/profile/user/wow/protected-character/1?namespace=profile-us",
            },
            user: {
                href: "https://us.api.blizzard.com/profile/user",
            },
            profile: {
                href: "https://us.api.blizzard.com/profile/user/wow?namespace=profile-us",
            },
        },
        id: 1,
        name: "Bigdawg",
        money: 1,
        character: {
            key: {
                href: "https://us.api.blizzard.com/profile/wow/character/barthilas/bigdawg?namespace=profile-us",
            },
            name: "bigdawg",
            id: 1,
            realm: {
                key: {
                    href: "https://us.api.blizzard.com/data/wow/realm/1?namespace=dynamic-us",
                },
                name: "Bathtub",
                id: 1,
                slug: "bathtub",
            },
        },
        protected_stats: {
            total_number_deaths: 1,
            total_gold_gained: 1337,
            total_gold_lost: 0,
            total_item_value_gained: 1337,
            level_number_deaths: 0,
            level_gold_gained: 1337,
            level_gold_lost: 0,
            level_item_value_gained: 1337,
        },
        position: {
            zone: {
                name: "Crusty pub",
                id: 1,
            },
            map: {
                name: "Australia",
                id: 1,
            },
            x: 1,
            y: -1,
            z: 10,
            facing: 1.5,
        },
        bind_position: {
            zone: {
                name: "Crusty pub tavern",
                id: 1,
            },
            map: {
                name: "Eastern Australia",
                id: 0,
            },
            x: -1,
            y: 400.5,
            z: 102,
            facing: 5.1,
        },
        wow_account: 1,
    };

    return (
        <Character>
            <WowCharacterImage
                character={
                    currentCharacter.hasOwnProperty("character") &&
                    currentCharacter.character.key.href
                }
            />
            <Text>
                <h3>{props.name}</h3>
                <h4>Realm: {props.name}</h4>

                <p>Level: {props.level}</p>

                <p>Class: {props.playableClass}</p>

                <p>Race: {props.playableRace}</p>

                <p>Gender: {props.gender}</p>

                <p>Faction: {props.faction}</p>
            </Text>
        </Character>
    );
};

export default WowCharacter;
