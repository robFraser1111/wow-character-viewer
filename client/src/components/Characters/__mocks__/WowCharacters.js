import React from "react";
import WowCharacter from "./WowCharacter";
import styled from "styled-components";

const Characters = styled.div`
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
    max-width: ${(props) => props.theme.laptop};

    @media (min-width: ${(props) => props.theme.tablet}) {
        &:after {
            content: "";
            width: 700px;
        }
    }
`;

const Message = styled.h2`
    color: ${(props) => props.theme.white};
    text-align: center;

    a {
        color: ${(props) => props.theme.blue}
    }

    a:hover, a:focus {
        color: ${(props) => props.theme.blueLight}
`;

const WowCharacters = () => {
    const currentCharacters = {
        _links: {
            self: {
                href: "https://us.api.blizzard.com/profile/user/wow?namespace=profile-us",
            },
            user: {
                href: "https://us.api.blizzard.com/profile/user",
            },
            profile: {
                href: "https://us.api.blizzard.com/profile/user/wow?namespace=profile-us",
            },
        },
        id: 1,
        wow_accounts: [
            {
                id: 1,
                characters: [
                    {
                        character: {
                            href: "https://us.api.blizzard.com/profile/wow/character/blackrock/character-1?namespace=profile-us",
                        },
                        protected_character: {
                            href: "https://us.api.blizzard.com/profile/user/wow/protected-character/1?namespace=profile-us",
                        },
                        name: "Character-1",
                        id: 1,
                        realm: {
                            key: {
                                href: "https://us.api.blizzard.com/data/wow/realm/1?namespace=dynamic-us",
                            },
                            name: "Somerock",
                            id: 1,
                            slug: "somerock",
                        },
                        playable_class: {
                            key: {
                                href: "https://us.api.blizzard.com/data/wow/playable-class/1?namespace=static-1-us",
                            },
                            name: "Rat Knight",
                            id: 1,
                        },
                        playable_race: {
                            key: {
                                href: "https://us.api.blizzard.com/data/wow/playable-race/1?namespace=static-1-us",
                            },
                            name: "Rat",
                            id: 1,
                        },
                        gender: {
                            type: "FEMALE",
                            name: "Female",
                        },
                        faction: {
                            type: "ALLIANCE",
                            name: "Alliance",
                        },
                        level: 1,
                    },
                    {
                        character: {
                            href: "https://us.api.blizzard.com/profile/wow/character/barthilas/bigdawg?namespace=profile-us",
                        },
                        protected_character: {
                            href: "https://us.api.blizzard.com/profile/user/wow/protected-character/2?namespace=profile-us",
                        },
                        name: "Bigdawg",
                        id: 2,
                        realm: {
                            key: {
                                href: "https://us.api.blizzard.com/data/wow/realm/2?namespace=dynamic-us",
                            },
                            name: "Bathtub",
                            id: 2,
                            slug: "bathtub",
                        },
                        playable_class: {
                            key: {
                                href: "https://us.api.blizzard.com/data/wow/playable-class/2?namespace=static-2-us",
                            },
                            name: "Monk",
                            id: 2,
                        },
                        playable_race: {
                            key: {
                                href: "https://us.api.blizzard.com/data/wow/playable-race/2?namespace=static-2-us",
                            },
                            name: "Pandaren",
                            id: 2,
                        },
                        gender: {
                            type: "MALE",
                            name: "Male",
                        },
                        faction: {
                            type: "ALLIANCE",
                            name: "Alliance",
                        },
                        level: 2,
                    },
                ],
            },
        ],
        collections: {
            href: "https://us.api.blizzard.com/profile/user/wow/collections?namespace=profile-us",
        },
    };

    console.log("Characters " + currentCharacters);

    if (currentCharacters.hasOwnProperty("wow_accounts")) {
        return (
            <Characters>
                {currentCharacters.wow_accounts[0].characters.map(
                    (character) => (
                        <WowCharacter
                            character={character.href}
                            key={character.id}
                            characterId={character.id}
                            realmId={character.realm.id}
                            name={character.name}
                            realm={character.realm.name}
                            level={character.level}
                            playableClass={character.playable_class.name}
                            playableRace={character.playable_race.name}
                            gender={character.gender.name}
                            faction={character.faction.name}
                        />
                    )
                )}
            </Characters>
        );
    } else if (
        currentCharacters === "error" ||
        typeof currentCharacters === "object"
    ) {
        return (
            <Message>
                Could not find any characters.
                <br />
                Try{" "}
                <a href={`${process.env.REACT_APP_DOMAIN}:5000/logout`}>
                    logging out
                </a>{" "}
                and choosing another <u>region</u>.
            </Message>
        );
    } else {
        return <Message>Loading...</Message>;
    }
};

export default WowCharacters;
