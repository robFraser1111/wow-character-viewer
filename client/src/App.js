import React, { useState, useEffect } from "react";
import useFetch from "./hooks/useFetch";
// import { Routes, Route, Link } from "react-router-dom";
import Header from "./components/Header";
import LoginForm from "./components/LoginForm";
import UserContext from "./store/context";
import WowCharacters from "./components/Characters/WowCharacters";
import Tabs from "./components/Tabs";
import WowMounts from "./components/Mounts/WowMounts";
import WowPets from "./components/Pets/WowPets";
import Footer from "./components/Footer";

import Bg1 from "./images/wow-background-01.jpg";
import Bg2 from "./images/wow-background-02.jpg";
import Bg3 from "./images/wow-background-03.jpg";
import Bg4 from "./images/wow-background-04.jpg";
import Bg5 from "./images/wow-background-05.jpg";
import Bg6 from "./images/wow-background-06.jpg";

import styled, { ThemeProvider } from "styled-components";

// Background images
const Bgs = [Bg1, Bg2, Bg3, Bg4, Bg5, Bg6];

// Set background image
const backgroundImage = Bgs[Math.floor(Math.random() * Bgs.length)];

const theme = {
    // Colours
    bgColor: "rgba(21, 23, 30, 0.7)",
    white: "#fefefe",
    brown: "#211510",
    yellow: "#f8b700",
    yellowDark: "#b1997f",
    yellowLight: "#ebdec2",
    blue: "#006fb2",
    blueDark: "#00629e",
    blueLight: "#007bc6",

    // Animation
    transition: "0.2s",

    // Breakpoints
    mobile: "320px",
    tablet: "768px",
    laptop: "1920px",
    desktop: "2560px",
};

const Wrapper = styled.div`
    min-height: 100vh;
    display: grid;
    grid-template-rows: 200px auto 100px;
    background-image: url(${backgroundImage});
    background-position: center;
    background-repeat: no-repeat;
    background-size: cover;
    background-color: ${theme.bgColor};
    background-blend-mode: color;
    background-attachment: fixed;
`;

const Main = styled.main`
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    margin-bottom: 100px;

    .hide {
        display: none;
    }
`;

const Section = styled.section`
    padding: 20px;
    flex: 2;
`;

function App() {
    const currentPlayer = useFetch("/api");

    const [region, setRegion] = useState("");
    const [locale, setLocale] = useState("");
    const [token, setToken] = useState("");
    const [tab, setTab] = useState("Characters");

    const buttons = ["Mounts", "Characters", "Pets"];

    // Set and Store Region and Locale in local storage so values don't get reset on page reload
    useEffect(() => {
        if (localStorage.getItem("region") === "") {
            localStorage.setItem("region", "us");
        }

        if (localStorage.getItem("locale") === "") {
            localStorage.setItem("locale", "en_US");
        }

        setRegion(localStorage.getItem("region"));
        setLocale(localStorage.getItem("locale"));
    }, []);

    // Manage the form inputs
    const regionHandler = (e) => {
        setRegion(e.target.value);
        localStorage.setItem("region", e.target.value);
    };
    const localeHandler = (e) => {
        setLocale(e.target.value);
        localStorage.setItem("locale", e.target.value);
    };

    // Manage the selected tabs
    const tabHandler = (e) => {
        setTab(e.target.value);
    };

    // Set token only if the call to the Api was successful
    useEffect(() => {
        if (currentPlayer.hasOwnProperty("token")) {
            setToken(currentPlayer.token);
        } else {
            return;
        }
    }, [currentPlayer]);

    return (
        <UserContext.Provider
            value={{
                token: token,
                region: region,
                locale: locale,
            }}
        >
            <ThemeProvider theme={theme}>
                <Wrapper>
                    <Header
                        heading="Character Viewer"
                        currentPlayer={currentPlayer}
                    />
                    <Main>
                        {token === "" && (
                            <LoginForm
                                regionHandler={regionHandler}
                                localeHandler={localeHandler}
                            />
                        )}

                        {token !== "" && (
                            <>
                                <Tabs
                                    tabHandler={tabHandler}
                                    buttons={buttons}
                                    currentTab={tab}
                                />
                                <Section
                                    className={
                                        tab === "Characters" ? "show" : "hide"
                                    }
                                >
                                    <WowCharacters />
                                </Section>
                                <Section
                                    className={
                                        tab === "Mounts" ? "show" : "hide"
                                    }
                                >
                                    <WowMounts />
                                </Section>
                                <Section
                                    className={tab === "Pets" ? "show" : "hide"}
                                >
                                    <WowPets />
                                </Section>
                            </>
                        )}
                    </Main>
                    <Footer />
                </Wrapper>
            </ThemeProvider>
        </UserContext.Provider>
    );
}

export default App;
