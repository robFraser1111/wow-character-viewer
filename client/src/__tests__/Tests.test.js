import React from "react";
import Footer from "../components/Footer";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import ReactDOM from "react-dom";
import App from "../App";
import useFetch from "../hooks/useFetch";
import WowCharacters from "../components/Characters/WowCharacters";
import WowCharacter from "../components/Characters/WowCharacter";
import WowCharacterImage from "../components/Characters/WowCharacterImage";
import Image from "../components/Characters/Image";

jest.mock("../hooks/useFetch");
jest.mock("../components/Characters/WowCharacters");
jest.mock("../components/Characters/WowCharacter");
jest.mock("../components/Characters/WowCharacterImage");
jest.mock("../components/Characters/Image");

let data = {};

beforeAll(() => {
    data = { test: "data" };
});

afterAll(() => {
    data = {};
});

test("App renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<App />, div);
});

test("Checks for Battle.net text in App", () => {
    render(<App />);
    expect(screen.getByText("Battle.net API")).toBeInTheDocument();
});

test("Footer renders correctly", () => {
    render(<Footer />);
    expect(screen.getByText("Rob Fraser")).toBeInTheDocument();
});

describe('WoW Characters', () => {
    test("Characters render with correct data", () => {
        const tree = renderer.create(<WowCharacters />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test("Character render with correct data", () => {
        const tree = renderer.create(<WowCharacter />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test("CharacterImage render with correct data", () => {
        const tree = renderer.create(<WowCharacterImage />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test("Image render with correct data", () => {
        const tree = renderer.create(<Image />).toJSON();
        expect(tree).toMatchSnapshot();
    });
})