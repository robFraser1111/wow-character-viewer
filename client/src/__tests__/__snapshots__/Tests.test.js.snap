// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WoW Characters Character render with correct data 1`] = `
<div
  className="sc-giYglK kgtxhA"
>
  <div
    className="sc-iqseJM hcnwtv"
  >
    <img
      src="https://render.worldofwarcraft.com/us/character/barthilas/69/1-main.jpg"
    />
  </div>
  <div
    className="sc-ezbkAF iugekb"
  >
    <h3 />
    <h4>
      Realm: 
    </h4>
    <p>
      Level: 
    </p>
    <p>
      Class: 
    </p>
    <p>
      Race: 
    </p>
    <p>
      Gender: 
    </p>
    <p>
      Faction: 
    </p>
  </div>
</div>
`;

exports[`WoW Characters CharacterImage render with correct data 1`] = `
<div
  className="sc-bYoBSM lbqQMy"
>
  <img
    src="https://render.worldofwarcraft.com/us/character/barthilas/69/1-main.jpg"
  />
</div>
`;

exports[`WoW Characters Characters render with correct data 1`] = `
<div
  className="sc-bqiRlB eXFHqo"
>
  <div
    className="sc-crHmcD cGlLJg"
  >
    <div
      className="sc-iqseJM hcnwtv"
    >
      <img
        src="https://render.worldofwarcraft.com/us/character/barthilas/69/1-main.jpg"
      />
    </div>
    <div
      className="sc-egiyK iLOulL"
    >
      <h3>
        Character-1
      </h3>
      <h4>
        Realm: 
        Character-1
      </h4>
      <p>
        Level: 
        1
      </p>
      <p>
        Class: 
        Rat Knight
      </p>
      <p>
        Race: 
        Rat
      </p>
      <p>
        Gender: 
        Female
      </p>
      <p>
        Faction: 
        Alliance
      </p>
    </div>
  </div>
  <div
    className="sc-crHmcD cGlLJg"
  >
    <div
      className="sc-iqseJM hcnwtv"
    >
      <img
        src="https://render.worldofwarcraft.com/us/character/barthilas/69/1-main.jpg"
      />
    </div>
    <div
      className="sc-egiyK iLOulL"
    >
      <h3>
        Bigdawg
      </h3>
      <h4>
        Realm: 
        Bigdawg
      </h4>
      <p>
        Level: 
        2
      </p>
      <p>
        Class: 
        Monk
      </p>
      <p>
        Race: 
        Pandaren
      </p>
      <p>
        Gender: 
        Male
      </p>
      <p>
        Faction: 
        Alliance
      </p>
    </div>
  </div>
</div>
`;

exports[`WoW Characters Image render with correct data 1`] = `
<img
  src="https://render.worldofwarcraft.com/us/character/barthilas/69/1-main.jpg"
/>
`;
